manager/state/raft/raft.go

```diff
@@ -340,7 +353,23 @@ func (n *Node) done() {
 // cluster state. If no saved state exists, it starts a single-node cluster.
 func (n *Node) Run(ctx context.Context) error {
 	ctx = log.WithLogger(ctx, logrus.WithField("raft_id", fmt.Sprintf("%x", n.Config.ID)))
-	defer n.done()
+	ctx, cancel := n.WithContext(ctx)
```

Comment:
I think it might be better to have cancellation earlier. Imagine we're in run-event-loop and we're blocking on some operation - we can't reach defer because it's blocking, so it's better to cancel it asap. Even if there is no such stuff now(I'm not quite sure, probably there is), it sounds like a plausible assumption.